app/jobs/purge_job.rb

```diff
@@ -41,8 +45,21 @@ def perform(*_args)
     ReportedCondition.where(assessment_id: Assessment.where(patient_id: Patient.where(purged: true).ids).ids).delete_all
     Assessment.where(patient_id: Patient.where(purged: true).ids).delete_all
 
-    # Send results
-    UserMailer.purge_job_email(purged, not_purged, eligible_count).deliver_now
+    # Gather statistics
+    job_info[:end_time] = DateTime.now
+    job_info[:not_purged_count] = not_purged.length
+    job_info[:purged_count] = purged.length
+    total_purged_emails_to_send = calculate_total_emails(job_info[:not_purged_count] + job_info[:purged_count])
+
+    # in_groups_of will not perform an iteration if the array is empty.
+    if (job_info[:not_purged_count] + job_info[:purged_count]).zero?
+      UserMailer.purge_job_email([], { current: 1, total: total_purged_emails_to_send }, job_info).deliver_later
+    else
+      # Send results in batches to avoid emails that are too large to send
+      (purged | not_purged).in_groups_of(ADMIN_OPTIONS['job_run_email_group_size'].to_i, false) do |group, index|
+        UserMailer.purge_job_email([group], { current: index, total: total_purged_emails_to_send }, job_info).deliver_later
+      end
```

Comment:
did you mean to do an `each_with_index` here? Right now the emails have text like "Message: {id: 12412} of 3".

![Screen Shot 2021-06-01 at 11 46 35 AM](https://user-images.githubusercontent.com/24628687/120368907-dacfc380-c2cf-11eb-9656-2b9859e8948a.png)

It looks like this would work if you change this section to
```ruby
      # Send results in batches to avoid emails that are too large to send
      (purged | not_purged).in_groups_of(ADMIN_OPTIONS['job_run_email_group_size'].to_i, false).each_with_index do |group, index|
        UserMailer.purge_job_email(group, { current: index + 1, total: total_purged_emails_to_send }, job_info).deliver_later
      end

```

Reply:
can confirm

![Screen Shot 2021-06-03 at 1 39 46 PM](https://user-images.githubusercontent.com/9956561/120688588-38e7dc80-c471-11eb-9caf-f5187282b5ad.png)

it also seems like the list of successfully purged monitorees is not being shown?

Reply:
@timwongj your secreen shot shows id 801 was purged?